### PR TITLE
ngx-datatable - stay current page after editing/deleting record. 

### LINKED ISSUE #17077: Angular: Hold current page when reloading data on datatable.
https://github.com/abpframework/abp/issues/17029

### PR DESCRIPTION
**Pull request by Sinan997**
### Ngx-datatable

Resolves #17077

When you change or delete something on ngx-datatable current page was resetting to 0. This pr makes current page persistent after edit/delete

### How to test it?

You can test it at these pages
- /identity/roles
- /identity/users

### CODE REVIEW

File: npm/ng-packs/packages/core/src/lib/services/list.service.ts [hunk 1]
```diff
@@ -121,12 +132,16 @@ export class ListService<QueryParamsType = ABP.PageQueryParams | any> implements
...
+    const skipCount = this._page * this._maxResultCount; // 20
+    if(skipCount === this._totalCount){
+      if(this.page === maxPage && this.page > 0){
+        this._skipCount = skipCount - this._maxResultCount
+        this.page = this.page - 1;
+      }
+    }else{
+      this._skipCount = skipCount;
+    }
   }
```
Comment (masum-ulu) [inline npm/ng-packs/packages/core/src/lib/services/list.service.ts:147]: What do you think about that ? @Sinan997 

```ts
  private resetPageWhenUnchanged(): void {
    const maxPage = Number((this._totalCount / this._maxResultCount).toFixed());
    const skipCount = this._page * this._maxResultCount;

    if (skipCount !== this._totalCount) {
      this._skipCount = skipCount;
      return;
    }

    if (this.page === maxPage && this.page > 0) {
      this._skipCount = skipCount - this._maxResultCount;
      this.page = this.page - 1;
    }
  }
  ```

File: npm/ng-packs/packages/theme-shared/extensions/src/lib/components/extensible-table/extensible-table.component.ts [hunk 2]
```diff
@@ -150,7 +150,7 @@ export class ExtensibleTableComponent<R = any> implements OnChanges {
 
   ngOnChanges({ data }: SimpleChanges) {
     if (!data?.currentValue) return;
-
+    this.list.totalCount = this.recordsTotal
```
Comment (mahmut-gundogdu) [inline npm/ng-packs/packages/theme-shared/extensions/src/lib/components/extensible-table/extensible-table.component.ts]: it seems code smells

**Review by mahmut-gundogdu — CHANGES_REQUESTED** [review]

File: npm/ng-packs/packages/theme-shared/extensions/src/lib/components/extensible-table/extensible-table.component.ts (hunk 2 above)
Comment (masum-ulu) [reply npm/ng-packs/packages/theme-shared/extensions/src/lib/components/extensible-table/extensible-table.component.ts]: 2 times sending request when the record created or deleted. We need to find an alternative solution

For now I found this

```ts
//extensible-table.component.ts

ngOnChanges({ data }: SimpleChanges) {
  if (!data?.currentValue) return;

  if (data.currentValue.length < 1) {
    this.list.totalCount = this.recordsTotal
  }
  ...
}
```

image

File: npm/ng-packs/packages/core/src/lib/services/list.service.ts [hunk 3]
```diff
@@ -121,12 +132,16 @@ export class ListService<QueryParamsType = ABP.PageQueryParams | any> implements
...
 
   private resetPageWhenUnchanged() {
-    const skipCount = this._page * this._maxResultCount;
-
-    if (skipCount === this._skipCount) {
-      this._page = 0;
-      this._skipCount = 0;
-    } else this._skipCount = skipCount;
+    const maxPage = Number(Number(this._totalCount / this._maxResultCount).toFixed()) // 2
```
Comment (masum-ulu) [inline npm/ng-packs/packages/core/src/lib/services/list.service.ts]: Why not using `totalCount`. Instead used `_totalCount` ?

**Comment by mahmut-gundogdu** [comment]
@gizemmutukurt  May you test it ? I think, I cant't

**Comment by gizemmutukurt** [comment]
error when copy-to-temp run

image

**Comment by mahmut-gundogdu** [comment]
I have rebase with dev. it must be solved. @gizemmutukurt  can yo retest  with delete branch in your local and do re-checkout. ?

**Review by gizemmutukurt — APPROVED** [review]

**Review by mahmut-gundogdu — APPROVED** [review]

**Pull request merged**